### ISSUE TITLE
Soften pydantic upper version constraint

### ISSUE THREAD

**Issue by noxan** [merge-api/merge-python-client#68]
is there any specific need to lock pydantic to `<2.5.0`? version `2.5.2` has already been released and it clashes with other packages. would seem more reasonable to me, locking pydantic to `<3.0.0`.

here a copy of the poetry error message:
```
Because no versions of mergepythonclient match >1.0.5,<2.0.0
 and mergepythonclient (1.0.5) depends on pydantic (>=1.9.2,<2.5.0), mergepythonclient (>=1.0.5,<2.0.0) requires pydantic (>=1.9.2,<2.5.0).
So, because conversa depends on both pydantic (^2.5.2) and mergepythonclient (^1.0.5), version solving failed.```

**Comment by lucasgadams** [comment]
Agree on this problem, merge is a small use case for use and the version locking on pydantic is blocking us from upgrading. Can we get some support here?

**Comment by dsinghvi** [comment]
Hey @lucasgadams @noxan -- thanks for reporting this issue. We'll get back to your shortly.

**Comment by lucasgadams** [comment]
Also, would be awesome if you can upgrade the sdk to fully use pydantic V2

**Comment by noxan** [comment]
@dsinghvi thanks for following up, you might want to check out https://github.com/merge-api/merge-python-client/pull/69

**Comment by noxan** [comment]
just saw it was resolved with #77 and release `1.0.6`, closing this issue and the pr therefore, thanks!
